pandas/tests/frame/methods/test_compare.py

```diff
@@ -238,17 +238,50 @@ def test_invalid_input_result_names(result_names):
         df1.compare(df2, result_names=result_names)
 
 
-def test_compare_ea_and_np_dtype():
-    # GH#44014
-    df1 = pd.DataFrame({"a": [4.0, 4], "b": [1.0, 2]})
-    df2 = pd.DataFrame({"a": pd.Series([1, pd.NA], dtype="Int64"), "b": [1.0, 2]})
-    result = df1.compare(df2, keep_shape=True)
+@pytest.mark.parametrize(
+    "ea_val,np_dtype_val",
+    [(4, pd.NA), (pd.NA, pd.NA), (pd.NA, 4)],
+)
+def test_compare_ea_and_np_dtype(ea_val, np_dtype_val):
+    ea = [4.0, ea_val]
```

Comment:
please add gh refs

please rename variable, ea stands for extension array

Reply:
will add gh refs

is arr ok?
```suggestion
    arr = [4.0, df1_val]
```

Reply:
yep works for me